distsql/distsql_test.go

```diff
@@ -407,406 +441,79 @@ func (r *mockResultSubset) MemSize() int64 { return int64(cap(r.data)) }
-				acodec.ReuseIntermChk(chk)
-			}
-			chk.Reset()
-		}
-	}
-}
-
-type dataBufferGen struct {
-	times int
-	data  [][]byte
-	cur   int
-}
+func createSelectNormal(batch, totalRows int, ctx sessionctx.Context) (*selectResult, []*types.FieldType) {
+	request, _ := (&RequestBuilder{}).SetKeyRanges(nil).
+		SetDAGRequest(&tipb.DAGRequest{}).
+		SetDesc(false).
+		SetKeepOrder(false).
+		SetFromSessionVars(variable.NewSessionVars()).
+		SetMemTracker(memory.NewTracker(stringutil.StringerStr("testSuite.createSelectNormal"),
+			ctx.GetSessionVars().MemQuotaDistSQL)).
+		Build()
 
-func (g *dataBufferGen) get() []byte {
-	if g.cur == len(g.data) {
-		g.cur, g.times = 0, g.times-1
-	}
-	if g.times == 0 {
-		return nil
+	/// 4 int64 types.
+	colTypes := []*types.FieldType{
+		{
+			Tp:      mysql.TypeLonglong,
+			Flen:    mysql.MaxIntWidth,
+			Decimal: 0,
+			Flag:    mysql.BinaryFlag,
+			Charset: charset.CharsetBin,
+			Collate: charset.CollationBin,
+		},
 	}
-	res := make([]byte, 0, 1024)
-	res = append(res[:0], g.data[g.cur]...)
-	return res
-}
+	colTypes = append(colTypes, colTypes[0])
+	colTypes = append(colTypes, colTypes[0])
+	colTypes = append(colTypes, colTypes[0])
 
-func populateBufferForChunk(numCols, numRows int) []byte {
-	colTypes := make([]*types.FieldType, numCols)
-	for i := 0; i < numCols; i++ {
-		colTypes[i] = &types.FieldType{Tp: mysql.TypeString}
-	}
-	chk := chunk.New(colTypes, numRows, numRows)
+	// Test Next.
+	var response SelectResult
+	response, _ = Select(context.TODO(), ctx, request, colTypes, statistics.NewQueryFeedback(0, nil, 0, false))
 
-	for rowOrdinal := 0; rowOrdinal < numRows; rowOrdinal++ {
-		for colOrdinal := 0; colOrdinal < numCols; colOrdinal++ {
-			chk.AppendString(colOrdinal, "123456")
-		}
-	}
-	codec := chunk.NewCodec(colTypes)
-	return codec.Encode(chk)
-}
+	result, _ := response.(*selectResult)
+	resp, _ := result.resp.(*mockResponse)
+	resp.total = totalRows
+	resp.batch = batch
 
-func buildBigResponse(gen *dataBufferGen) {
-	gen.data = make([][]byte, 20)
-	for i := 0; i < 20; i++ {
-		gen.data[i] = make([]byte, 0, 1024)
-	}
-	betchSize := 32
-	for i := 0; i < 20; i++ {
-		gen.data[i] = populateBufferForChunk(4, betchSize)
-		if betchSize < 1024 {
-			betchSize *= 2
-		}
-	}
+	return result, colTypes
 }
 
-func BenchmarkDecodeToChunkRequiredRows_BigResponse(b *testing.B) {
-	datagen := &dataBufferGen{}
-	buildBigResponse(datagen)
-
-	colTypes := make([]*types.FieldType, 4)
-	for i := 0; i < 4; i++ {
-		colTypes[i] = &types.FieldType{Tp: mysql.TypeString}
-	}
-	chk := chunk.New(colTypes, 4, 1024)
-	acodec := chunk.NewDecoder(
-		chunk.NewChunkWithCapacity(colTypes, 0),
-		colTypes)
-
-	chk.SetRequiredRows(100, 1024)
-	chk.Reset()
-	datagen.times = b.N
-	datagen.cur = 0
-	b.ResetTimer()
-
+func BenchmarkSelectResponseChunk_BigResponse(b *testing.B) {
+	s := &testSuite{}
+	s.SetUpSuite(nil)
+	b.StopTimer()
+	selectResult, colTypes := createSelectNormal(4000, b.N*4000, s.sctx)
```

Comment:
We should not use b.N here, this will falsify the result of benchmark.

Try to benchmark in normal way,
```go
for i := 0; i < b.N; i++ {
    // prepare
    b.StartTimer()
    for true {
        ...
    }
}
```

Reply:
Addressed. PTAL again. @SunRunAway